src/main/java/com/mraof/minestuck/block/EnumCassetteType.java

```diff
@@ -1,24 +1,88 @@
 package com.mraof.minestuck.block;
 
+import com.mraof.minestuck.util.MSSoundEvents;
+import net.minecraft.sounds.SoundEvent;
+import net.minecraft.sounds.SoundEvents;
 import net.minecraft.util.StringRepresentable;
+import net.minecraft.world.effect.MobEffectInstance;
+import net.minecraft.world.effect.MobEffects;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Objects;
+import java.util.function.Supplier;
 
 public enum EnumCassetteType implements StringRepresentable
 {
-	NONE,
-	THIRTEEN,
-	CAT,
-	BLOCKS,
-	CHIRP,
-	FAR,
-	MALL,
-	MELLOHI,
-	EMISSARY_OF_DANCE,
-	DANCE_STAB_DANCE,
-	RETRO_BATTLE_THEME;
+	NONE(() -> null, null),
+	ELEVEN(() -> SoundEvents.MUSIC_DISC_11, onHitEffect(() -> new MobEffectInstance(MobEffects.WITHER, 80, 0),
+			0.10F)),
+	THIRTEEN(() -> SoundEvents.MUSIC_DISC_13, onHitEffect(() -> new MobEffectInstance(MobEffects.HUNGER, 100, 0), 0.30F)),
+	BLOCKS(() -> SoundEvents.MUSIC_DISC_BLOCKS, onHitEffect(() -> new MobEffectInstance(MobEffects.CONFUSION, 100, 0),
+			0.15F)),
+	CAT(() -> SoundEvents.MUSIC_DISC_CAT, userEffect(() -> new MobEffectInstance(MobEffects.NIGHT_VISION, 100, 0,
+			false, false, false))),
+	CHIRP(() -> SoundEvents.MUSIC_DISC_CHIRP, userEffect(() -> new MobEffectInstance(MobEffects.SLOW_FALLING, 100, 0,
+			false, false, false))),
+	DANCE_STAB_DANCE(MSSoundEvents.MUSIC_DISC_DANCE_STAB_DANCE, userEffect(() -> new MobEffectInstance(MobEffects.DAMAGE_BOOST, 100, 0,
+			false, false, false))),
+	EMISSARY_OF_DANCE(MSSoundEvents.MUSIC_DISC_EMISSARY_OF_DANCE, onHitEffect(() -> new MobEffectInstance(MobEffects.POISON, 400, 0),
+			0.1F)),
+	FAR(() -> SoundEvents.MUSIC_DISC_FAR, userEffect(() -> new MobEffectInstance(MobEffects.MOVEMENT_SPEED, 100, 0,
+			false, false, false))),
+	MALL(() -> SoundEvents.MUSIC_DISC_MALL, userEffect(() -> new MobEffectInstance(MobEffects.WATER_BREATHING, 100, 0,
+			false, false, false))),
+	MELLOHI(() -> SoundEvents.MUSIC_DISC_MELLOHI, onHitEffect(() -> new MobEffectInstance(MobEffects.LEVITATION, 60, 0),
+			0.20F)),
+	PIGSTEP(() -> SoundEvents.MUSIC_DISC_PIGSTEP, userEffect(() -> new MobEffectInstance(MobEffects.FIRE_RESISTANCE, 100, 0,
+			false, false, false))),
+	RETRO_BATTLE_THEME(MSSoundEvents.MUSIC_DISC_RETRO_BATTLE_THEME, userEffect(() -> new MobEffectInstance(MobEffects.DAMAGE_RESISTANCE, 100, 0,
+			false, false, false))),
+	STAL(() -> SoundEvents.MUSIC_DISC_STAL, userEffect(() -> new MobEffectInstance(MobEffects.JUMP, 100, 1,
+			false, false, false))),
+	STRAD(() -> SoundEvents.MUSIC_DISC_STRAD, onHitEffect(() -> new MobEffectInstance(MobEffects.UNLUCK, 200, 0),
+			0.10F)),
+	WAIT(() -> SoundEvents.MUSIC_DISC_WAIT, onHitEffect(() -> new MobEffectInstance(MobEffects.MOVEMENT_SLOWDOWN, 200, 0),
+			0.30F)),
+	WARD(() -> SoundEvents.MUSIC_DISC_WARD, onHitEffect(() -> new MobEffectInstance(MobEffects.GLOWING, 750, 0), 1F));
```

Comment:
Still trying to gauge the impact of these effects, but I have this to say for now. I think the following effects need some changes: 
-Blocks: Suggestion to change effect. The confusion/nausea effect can be actually disorienting to the human playing the game. There is a video settings slider that allows people to control the strength of the distortion, however its also not something people may be expecting to come up against, and if they do reduce the distortion than having it on the weapon doesnt accomplish much
-Dance Stab Dance: Suggestion to change effect. While the song is being played, the damage boost effects means the weapon essentially deals 13 damage which is more than even the top tier swords
-Emissary of Dance: Consider reducing the length of the poison damage. Another weapon which deals poison damage(Red Eye) has a duration of 140 which is less than half the 400 the cassette is currently set to
-Far: essentially permanent speed buff may be too strong
-Retro Battle Theme: essentially permanent damage resistance may be too strong